### PR TITLE
Fix poco discover server users

### PR DESCRIPTION
**Pull request by MrPetovan**
Fixes https://github.com/friendica/friendica/issues/6386#issuecomment-461799483

### CODE REVIEW

File: src/Protocol/PortableContact.php [hunk 1]
```diff
@@ -1645,21 +1650,19 @@ private static function discoverFederation()
...
-		$server = $r[0];
-
 		// Discover new servers out there (Works from Friendica version 3.5.2)
 		self::fetchServerlist($server["poco"]);
 
 		// Fetch all users from the other server
 		$url = $server["poco"] . "/?fields=displayName,urls,photos,updated,network,aboutMe,currentLocation,tags,gender,contactType,generation";
 
-		Logger::log("Fetch all users from the server " . $server["url"], Logger::DEBUG);
+		Logger::info("Fetch all users from the server " . $server["url"]);
```
Comment (nupplaphil) [inline src/Protocol/PortableContact.php]:
```suggestion
		Logger::info("Fetch all users from the server.", ['url' => $server["url"]]);
```

File: src/Protocol/PortableContact.php [hunk 2]
```diff
@@ -1687,16 +1690,16 @@ public static function discoverSingleServer($id)
 				$curlResult = Network::curl($url);
 
 				if ($curlResult->isSuccess() && !empty($curlResult->getBody())) {
-					Logger::log("Fetch all global contacts from the server " . $server["nurl"], Logger::DEBUG);
+					Logger::info("Fetch all global contacts from the server " . $server["nurl"]);
```
Comment (nupplaphil) [inline src/Protocol/PortableContact.php]:
```suggestion
					Logger::info("Fetch all global contacts from the server.", ['nurl' => $server["nurl"]]);
```

File: src/Protocol/PortableContact.php [hunk 3]
```diff
@@ -1687,16 +1690,16 @@ public static function discoverSingleServer($id)
...
 
 					if (!empty($data)) {
 						$success = self::discoverServer($data);
 					}
 				}
 
-				if (!$success && (Config::get('system', 'poco_discovery') > 2)) {
-					Logger::log("Fetch contacts from users of the server " . $server["nurl"], Logger::DEBUG);
+				if (!$success && !empty($data) && Config::get('system', 'poco_discovery') >= self::USERS_GCONTACTS_FALLBACK) {
+					Logger::info("Fetch contacts from users of the server " . $server["nurl"]);
```
Comment (nupplaphil) [inline src/Protocol/PortableContact.php]:
```suggestion
					Logger::info("Fetch contacts from users of the server.", ['nurl' => $server["nurl"]]);
```

File: src/Protocol/PortableContact.php [hunk 4]
```diff
@@ -1670,7 +1673,7 @@ public static function discoverSingleServer($id)
 				self::discoverServer($data, 2);
 			}
 
-			if (Config::get('system', 'poco_discovery') > 1) {
+			if (Config::get('system', 'poco_discovery') >= self::USERS_GCONTACTS) {
```
Comment (nupplaphil) [inline src/Protocol/PortableContact.php]: Wouldn't be a binary "has" better? It seems like `>=` is used instead of "has" GCONTACTS and GCONTACTS_FALLBACK

**Comment by MrPetovan** [comment]
Sorry @nupplaphil, I didn't take the time to address your reviews, I will keep the context array for logging in the future, and I probably will revisit the Portable Contact constants at some point in the future..

**Pull request merged**